sopel/tools/time.py

```diff
@@ -200,68 +252,92 @@ def format_time(db=None, config=None, zone=None, nick=None, channel=None,
     """
-    utc = pytz.timezone('UTC')
-    tformat = None
+    target_tz: datetime.tzinfo = pytz.utc
+    tformat: Optional[str] = None
+
+    # get an aware datetime
+    if not time:
+        time = pytz.utc.localize(datetime.datetime.utcnow())
+    elif not time.tzinfo:
+        time = pytz.utc.localize(time)
+
+    # get target timezone
+    if zone:
+        target_tz = pytz.timezone(zone)
+
+    # get format for nick or channel
     if db:
         if nick:
             tformat = db.get_nick_value(nick, 'time_format')
         if not tformat and channel:
             tformat = db.get_channel_value(channel, 'time_format')
+
+    # get format from configuration
     if not tformat and config and config.core.default_time_format:
         tformat = config.core.default_time_format
+
+    # or default to hard-coded format
     if not tformat:
         tformat = '%Y-%m-%d - %T %z'
 
-    if not time:
-        time = datetime.datetime.now(tz=utc)
-    elif not time.tzinfo:
-        time = utc.localize(time)
-
-    if not zone:
-        zone = utc
-    else:
-        zone = pytz.timezone(zone)
-
-    return time.astimezone(zone).strftime(tformat)
+    # format local time with format
+    return time.astimezone(target_tz).strftime(tformat)
 
 
-def seconds_to_split(seconds):
+def seconds_to_split(seconds: int) -> Duration:
     """Split an amount of ``seconds`` into years, months, days, etc.
 
-    :param int seconds: amount of time in seconds
-    :return: the time split into a tuple of years, months, days, hours,
+    :param seconds: amount of time in seconds
+    :return: the time split into a named tuple of years, months, days, hours,
              minutes, and seconds
-    :rtype: :class:`tuple`
 
     Examples::
 
         >>> seconds_to_split(7800)
-        (0, 0, 0, 2, 10, 0)
+        Duration(years=0, months=0, days=0, hours=2, minutes=10, seconds=0)
         >>> seconds_to_split(143659)
-        (0, 0, 1, 15, 54, 19)
+        Duration(years=0, months=0, days=1, hours=15, minutes=54, seconds=19)
 
     .. versionadded:: 7.1
+
+    .. versionchanged:: 8.0
+
+        This function returns a :class:`Duration` named tuple.
+
     """
     years, seconds_left = divmod(int(seconds), YEARS)
     months, seconds_left = divmod(seconds_left, MONTHS)
     days, seconds_left = divmod(seconds_left, DAYS)
     hours, seconds_left = divmod(seconds_left, HOURS)
     minutes, seconds_left = divmod(seconds_left, MINUTES)
 
-    return years, months, days, hours, minutes, seconds_left
-
-
-def get_time_unit(years=0, months=0, days=0, hours=0, minutes=0, seconds=0):
+    return Duration(years, months, days, hours, minutes, seconds_left)
+
+
+def get_time_unit(
+    years: int = 0,
+    months: int = 0,
+    days: int = 0,
+    hours: int = 0,
+    minutes: int = 0,
+    seconds: int = 0,
+) -> Tuple[
+    Tuple[int, str],
+    Tuple[int, str],
+    Tuple[int, str],
+    Tuple[int, str],
+    Tuple[int, str],
+    Tuple[int, str],
+]:
```

Comment:
Monster return value type-hint there, but is it worth creating another custom type like `Duration`? Survey says probably not.